src/mmcmp.cpp

```diff
@@ -185,15 +185,20 @@ BOOL MMCMP_Unpack(LPCBYTE *ppMemFile, LPDWORD pdwMemLength)
 		LPMMCMPSUBBLOCK psubblk;
 
 		if (dwMemPos >= dwMemLength - 20) break;
-		memcpy(tmp1, lpMemFile+dwMemPos, 28);
 		pblk = (LPMMCMPBLOCK)(tmp1);
 		psubblk = (LPMMCMPSUBBLOCK)(tmp1+20);
+
+		memcpy(pblk, lpMemFile+dwMemPos, 20);
 		swap_block(pblk);
-		swap_subblock(psubblk);
```

Comment:
mmcmp.c in libmikmod and libxmp _might_ need revisiting, too.

Reply:
Both look good to me. MikMod uses a sane file IO abstraction that shouldn't have the same problem this did. libxmp's bounds checks are built into the `fread` calls it makes to load the blocks/sub blocks.

Reply:
I preemptively checked MikMod and libxmp for the MMCMP bounds check cleanup I just pushed, and it looks like neither are affected by the bizarre/redundant bounds checking that libmodplug had in place either (MikMod checks writes against an end pointer rather than bounding `unpk_size`; libxmp writes to stdio).

Reply:
Thanks for checking them. I remember having tried to minimize load times in libmikmod but our versions diverged and libmodplug seems to have lost interest so I haven't submitted any more patches for it.

(I pinged million times for my patches here, also sent private email, no result. Not sure what you will get with this patchset, but I will merge it in my fork and to SDL_mixer and SDL_sound's own sources.)

Reply:
OK. Fuzzing has slowed down significantly at finding new bugs, so I'll probably squash/reorganize this patch soon so it's ready to be applied elsewhere. Since there have been so many changes, I think I'll do a commit for each loader instead of a single commit.